### ISSUE TITLE
After updating playlist the order of videos is incorrect

### ISSUE THREAD

**Issue by iG8R** [axcore/tartube#476]
Windows 10 (21H1, 19043.1415)
Tartube v2.4.165

The title says for itself.
Its moved to a separate issue from here [https://github.com/axcore/tartube/issues/472#issuecomment-1335975224](https://github.com/axcore/tartube/issues/472#issuecomment-1335975224)
Playlist [https://vk.com/video/playlist/-24440848_54694294](https://vk.com/video/playlist/-24440848_54694294)
After updating playlist the order of videos is incorrect.
<details><summary>Screenshot</summary>

![Tartube-Sorting-Video_2022-12-02_235012](https://user-images.githubusercontent.com/11407417/205408174-68f80ecd-ae79-49eb-b151-a724a27db6bf.gif)

</details>

**Comment by iG8R** [comment]
>Please "check" the playlist, and then show me the yellow text at the top of the Output tab. Then I will try to reproduce your exact problem.

![Tartube-Sorting-Yellow-text-2022-12-04_202948](https://user-images.githubusercontent.com/11407417/205508869-ba4519b9-6421-4dae-b29c-4d08f98b1602.jpg)

`yt-dlp --newline -i --hls-prefer-native --write-description --cookies E:/TarTube(Downloader)/cookies.txt --write-thumbnail --windows-filenames --output E:/TarTube(Downloader)/Расколотая битвой синева небес Целый год/%(title)s.%(ext)s --dump-json https://vk.com/video/playlist/-24440848_54694294`


>Try adding https://vk.com/video/playlist/-24440848_54694294 as a channel (not a playlist). Do you still see the same problem? (Tartube tries to sort playlists so the order is the same, as the order on the video website.)

If now I add the playlist either as a channel or as a playlist(with a different name), the order is correct.
The point is that the sorting issue occurred after the new video was added to the playlist. To properly reproduce this issue, need to add the playlist now and wait for a new video to appear.

**Comment by iG8R** [comment]
Tartube ver 2.4.173
You can see from the video below what I meant in my previous saying.
Also from the video it could be seen some bugs with the UI.
1. "Show tooltips for videos..." is buggy. 
2. Is it possible to make "Size" with "0" value not be able to block realtime view in the View window during checking and downloading?

GIF animation duration: 2min20sec

![python3_rze5BGqy3e](https://user-images.githubusercontent.com/11407417/205513920-9148f253-c476-47b8-ad01-3e9a054ff2d5.gif)

**Comment by iG8R** [comment]
Another instance of the issue.
[https://vk.com/video/playlist/-179915138_55](https://vk.com/video/playlist/-179915138_55)

Yesterday 2022-12-04 at about 13:00 to the playlist on the site was added new episode (87).
Also yesterday 2022-12-04 at 02:15 I manually added 2 missing episodes (7 and 10).

![Tartube-sorting-2022-12-05_013549](https://user-images.githubusercontent.com/11407417/205523355-f07de3ba-ecef-4bdd-87b2-56107cde42b2.jpg)
![Tartube-sorting-2022-12-05_012108-2](https://user-images.githubusercontent.com/11407417/205523372-b762acdd-8d54-4807-a1d9-71dd880de2c0.jpg)

**Comment by iG8R** [comment]
At last the issue becomes clearer.
The playlist on VK is updated.
After rechecking the playlists (old and new), the sorting problem remained, but after rechecking the channel, there is no sorting issue.
![Tartube-sorting-2022-12-06_220042](https://user-images.githubusercontent.com/11407417/206012663-c9042101-37b8-486a-aad6-ecddf46c3cb9.jpg)
![Tartube-sorting-2022-12-06_220224](https://user-images.githubusercontent.com/11407417/206012680-91128777-55a3-49ce-b5a0-b819437c90ea.jpg)
![Tartube-sorting-2022-12-06_220337](https://user-images.githubusercontent.com/11407417/206012706-5650bc3b-32d5-4309-8acc-3b603b6e0418.jpg)

**Comment by axcore** [comment]
I have tested https://vk.com/video/playlist/-24440848_54694294 on both Linux and MS Windows, using identical download options to yours, but I can't reproduce your problem.

You say that it's necessary to wait for a new video to be added to the playlist. I made a backup of my database, so I can test this playlist again, when the next video is added.

I don't know what the cause of your problem is, but I suspect VK renames each video's playlist number, when a new video is added to the playlist. If two videos both have the same playlist number, then they might be sorted in a random order. I will test that when we have a new video.

**Comment by axcore** [comment]
In v2.4.190:

1. The tooltips problem is fixed
2. The page size problem is fixed. You can't set the size to 0 any more, actually you can't set the size to -1 either ;)

**Comment by axcore** [comment]
3. I tweaked the sorting algorithm, so that two videos with the same playlist number are not sorted randomly. But I don't know yet, if it will help your problem.

**Comment by iG8R** [comment]
Thank you, I'll check it next time when the playlist is updated.

**Comment by iG8R** [comment]
Thanks a lot, the issue with incorrect sorting is gone:)
Regarding point number 2 - maybe, it'd has been more convenient for the Tartube usability to make "0" as the equivalent of no limits on the number of items on the page?

**Comment by axcore** [comment]
It's not a good idea. If 100 videos are visible, Tartube must load 100 thumbnails from random places on the hard drive. Now imagine how long it will take to download 1000 thumbnails, or 10000 thumbnails.

**Comment by iG8R** [comment]
I definitely agree with you, but you already did what you could by setting this parameter to 50 from the initial installation.
It might be worth letting the user choose the desired parameter and setting it to 999999 is less pretty than 0 when they want an unlimited number of elements on the page?
And they don't care about 1000 thumbnails as their 128-core CPU with 1TB RAM and 100TB Optane SSD can afford it:)

**Comment by axcore** [comment]
I considered using a 0 setting, but I think users need to be discouraged from creating grids of infinite size. It doesn't matter if you have a 128-core CPU, if the thumbnails are in different locations on your hard drive, then it will take some time to physically move the reader from one side of the spinning disc to the other.

**Comment by iG8R** [comment]
Eh, the sorting issue appeared again.
Tartube v2.4.198

![Tartube-order-issue-2022-12-22_043832](https://user-images.githubusercontent.com/11407417/209054384-4b4acf6a-ebe3-46fd-9188-0c1044c66465.jpg)
![Tartube-order-channel-2022-12-22_044037](https://user-images.githubusercontent.com/11407417/209054394-b87b4257-7c24-4fbc-833f-d9650aee0e2e.jpg)

**Comment by axcore** [comment]
I tested those playlists today, and I don't see the same problems.

![vk1](https://user-images.githubusercontent.com/37951926/211187506-f22920b9-83b1-4e39-8cfc-486fddcebf45.png)
![vk2](https://user-images.githubusercontent.com/37951926/211187508-490a3fae-e80c-4033-8c0b-b2b3272d941b.png)

**Comment by iG8R** [comment]
After initially switching to the playlist, the order is fine, but switching again makes that playlist messy again.

[Tartube-Video-Order-Video_2023-01-08_140042.webm](https://user-images.githubusercontent.com/11407417/211195312-0a678846-4612-444e-8f1f-cd2a3ae32243.webm)

**Comment by iG8R** [comment]
This playlist https://vk.com/video/playlist/-179915138_101 is weird even more regarding the order of the videos:
![Tartube-Order-2023-01-08_172103](https://user-images.githubusercontent.com/11407417/211205160-2baab2c5-9292-4ebf-a9dd-97b0a5b6d863.jpg)

**Comment by iG8R** [comment]
I'll leave my database for you, maybe something is wrong with it.
[tartube.zip](https://github.com/axcore/tartube/files/10368794/tartube.zip)

**Comment by axcore** [comment]
For some reason, your **Ideal world** channel was not sorted properly. I can't explain why, because sorting happens frequently. Anyway, I have added an extra sort, whenever videos in the Videos tab are completely re-drawn (for example, when you click on a channel/playlist/folder on the left side).

I also tweaked the sorting algorithm so it handles VK's weirdness better.

By the way, v2.4.221 should also solve your grid width problem. The reason some grid tiles were bigger than the visible window, is that your channels have very long names. I tweaked the grid code to handle that.
